docs/dev/content/implementation.rst

```diff
@@ -0,0 +1,116 @@
+Implementation Details and Workflows
+====================================
+
+To achieve using separated database for each channel and being able to switch channel dynamically, the following date structure and utility functions have been implemented.
+
+ContentDBRoutingMiddleware
+--------------------------
+
+This middleware will get used on every request sent to content app, and every response content app sends out.
```

Comment:
Tweak to: "sent to the content app API"

Actually, it will be applied to any request (there's no easy way to have it run just on requests to a single app). So, perhaps this would be better as:
`This middleware will be applied to every request, and will dynamically select a database based on the channel_id.`